pkg/auth/dependency/mfa/pq/store.go

```diff
@@ -308,6 +317,51 @@ func (s *storeImpl) CreateTOTP(a *mfa.TOTPAuthenticator) error {
 	return nil
 }
 
+func (s *storeImpl) GetTOTP(userID string, id string) (*mfa.TOTPAuthenticator, error) {
+	q1 := s.sqlBuilder.Tenant().
+		Select(
+			"a.id",
+			"a.user_id",
+			"a.type",
+			"at.activated",
+			"at.created_at",
+			"at.activated_at",
+			"at.secret",
+			"at.display_name",
+		).
+		From(s.sqlBuilder.FullTableName("authenticator"), "a").
+		Join(
+			s.sqlBuilder.FullTableName("authenticator_totp"),
+			"at",
+			"a.id = at.id",
+		).
+		Where("a.user_id = ? AND a.id = ?", userID, id)
+
+	row := s.sqlExecutor.QueryRowWith(q1)
+	var a mfa.TOTPAuthenticator
+	err := s.scanTOTPAuthenticator(row, &a)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			err = mfa.ErrAuthenticatorNotFound
+		}
+		return nil, err
+	}
+	return &a, nil
+}
+
+func (s *storeImpl) UpdateTOTP(a *mfa.TOTPAuthenticator) error {
```

Comment:
Personally I'll make it takes 3 arguments (id, activated, activatedAt), since it does not update other fields in the authenticator.

Reply:
Taking the whole struct seems making the API more pretty. We have Create to return a pointer to the struct, Update to take a pointer and Delete to take a pointer.

Reply:
I'm worried that later on people may see `UpdateTOTP` and use it, thinking that it would update other fields (e.g. display name). Especially when godoc just states `UpdateTOTP updates TOTP authenticator`, it is misleading.

Reply:
Let me update the documentation to better describe the behavior.